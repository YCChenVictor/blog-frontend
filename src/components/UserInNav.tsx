import React from "react";
import SignUpLogin from "./SignUpLogin";

interface UserInNavProps {
  loggedIn: boolean;
}

const UserInNav: React.FC<UserInNavProps> = ({ loggedIn }) => {
  return (
    <>
      {loggedIn ? (
        <>
          <SignUpLogin />
          {/* going to add logout */}
        </>
      ) : (
        <>
          <SignUpLogin />
        </>
      )}
    </>
  );
};

export default UserInNav;
