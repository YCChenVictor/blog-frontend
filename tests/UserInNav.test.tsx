import * as renderer from "react-test-renderer";
import * as React from "react";
import UserInNav from "../src/components/UserInNav";

describe("UserInNav", () => {
  it("renders UserInNav component without crashing", () => {
    const component = renderer.create(<UserInNav loggedIn={false} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
