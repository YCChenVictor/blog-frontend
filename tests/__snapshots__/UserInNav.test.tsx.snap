// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInNav renders UserInNav component without crashing 1`] = `
<div>
  <div
    className="top-4 right-4 rounded-sm"
  >
    <button
      className="px-4 py-2 text-white bg-gray-700 rounded-lg hover:bg-gray-800 focus:outline-none"
      onClick={[Function]}
    >
      Login
    </button>
    <button
      className="px-4 py-2 text-white bg-gray-700 rounded-lg hover:bg-gray-800 focus:outline-none"
      onClick={[Function]}
    >
      Signup
    </button>
  </div>
</div>
`;
